### ISSUE TITLE
rancher2_auth_config_github should not try to log me in

### ISSUE THREAD

**Issue by drpebcak** [rancher/terraform-provider-rancher2-archive#66]
```
# Create a new rancher2 Auth Config Github
resource "rancher2_auth_config_github" "github" {
  client_id = "<GITHUB_CLIENT_ID>"
  client_secret = "<GITHUB_CLIENT_SECRET>"
  code = "<GITHUB_AUTH_CODE>" # This is only for UI flow to authenticate an actual user
}
```

Error from `terraform apply` that helps show what it is trying to do:
```
* rancher2_auth_config_github.github: [ERROR] Posting Auth Config github: Bad response statusCode [500]. Status [500 Internal Server Error]. Body: [message=server error while authenticating: Received Error from github bad_verification_code, description from github The code passed is incorrect or expired., baseType=error, code=ServerError] from [https://rancher.example.com/v3/githubConfigs/github?action=testAndApply]
```

We shouldn't require `code` that is used for oauth authentication flow when working through terraform. There is no user to login when running this through the api. You also can't really get the value for `code` unless you manually set up oauth first.

Can we just set up the oauth configuration without trying to authenticate a user?

**Comment by cloudnautique** [comment]
TestAndApply, should be skipped from an automated workflow, I would imagine we just save the config as is. The purpose in the UI is to login and set that user as an Admin. In this flow, I would expect that no additional admin users would be set. The local admin will still be required to add additional admins.

**Comment by drpebcak** [comment]
Just noticed the same thing applies to Azure AD. The fix for both of these should be the same - just save the settings, don't actually try to go through the auth flow or set a user from that external provider as admin.

**Comment by rawmind0** [comment]
Fix on PR #61

**Comment by cloudnautique** [comment]
Closing.
